src/mds/SimpleLock.cc

```diff
@@ -0,0 +1,43 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*- 
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2004-2006 Sage Weil <sage@newdream.net>
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software 
+ * Foundation.  See file COPYING.
+ * 
+ */
+
+
+#include "SimpleLock.h"
+#include "Mutation.h"
+
+void SimpleLock::dump(Formatter *f) const {
+  assert(f != NULL);
+  if (is_sync_and_unlocked()) {
```

Comment:
What I would like to be doing here is outputting null where no lock is taken.  Our formatter layer doesn't permit that, so outputting {} is the next best thing (and most languages consuming this will evaluate it as false, so it's convenient enough).  The thing I'm really allergic to is having individual fields (including their keys) appear and disappear, rather than having a whole object be absent/present.